src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs

```diff
@@ -93,14 +93,39 @@ public void UpdateVariable(string name, decimal value)
 
         public void UpdateVariable(string name, int value)
         {
-            UpdateVariable(name, new NumberValue(IRContext.NotInSource(FormulaType.Number), value));
+            UpdateVariable(name, new DecimalValue(IRContext.NotInSource(FormulaType.Decimal), value));
         }
 
         public void UpdateVariable(string name, long value)
         {
             UpdateVariable(name, new DecimalValue(IRContext.NotInSource(FormulaType.Decimal), value));
         }
 
+        public void UpdateVariable(string name, string value)
+        {
+            UpdateVariable(name, new StringValue(IRContext.NotInSource(FormulaType.String), value));

```

Comment:
We should call the FormulaValue.New() overloads. That more closely mimics the public surface. 

(separately, we should get rid of IRContext)

Reply:
Great call.  Done.  Which led to many more changes, but also some more bugs being discovered.